bwe/RateController.cpp

```diff
@@ -112,9 +167,14 @@ void RateController::onSenderReportSent(uint64_t timestamp, uint32_t ssrc, uint3
             _probe.targetQueue,
             _model.queue.size(),
             reportNtp);
-        if (_probe.count == 10 || _probe.count == 30)
+
+        const auto probesSinceLastReduction = _probe.count - _probe.countOnLastIntervalReduction;
+
+        if (_probe.interval < Probe::MAX_INTERVAL && (probesSinceLastReduction == 10 || probesSinceLastReduction == 30))
         {
-            _probe.interval *= 2;
+            // New variable to shut up compiler warnings on mac
+            constexpr auto maxInterval = Probe::MAX_INTERVAL;
```

Comment:
Actually the comment is wrong. it was not a compiler warning but linker error. It says it can find the MAX_INTERVAL when I give it as parameter of `std::min`. `MAX_INTERVAL` is a `constexpr` and the value is assign in declaration, I am being used this a lot in C++17 and it works well but in C++14 a `static constexpr` member is not inlined (at least not in all cases). When I assign directly or use id a `if` condition, the compiler will use the literal value directly instead of reference something to be solved by the linker. But when I give it as an argument the compiler it will reference some entry to be linked.
But I agree the type should be explicitly to the be easiest to read the code since it is not explicit by the rhs of the assign.